### PR TITLE
Imx8 documentation

### PR DESCRIPTION
**Pull request by neilsh-msft**

### CODE REVIEW

File: Documentation/build-arm64-firmware.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    $ sudo apt-get install build-essential python python-dev python-crypto python-wand device-tree-compiler bison flex swig iasl uuid-dev wget git bc libssl-dev zlib1g-dev python3-pip
+    $ pushd ~
+    $ wget https://releases.linaro.org/components/toolchain/binaries/7.2-2017.11/aarch64-linux-gnu/gcc-linaro-7.2.1-2017.11-x86_64_aarch64-linux-gnu.tar.xz
+    $ tar xf gcc-linaro-7.2.1-2017.11-x86_64_aarch64-linux-gnu.tar.xz
+    $ rm gcc-linaro-7.2.1-2017.11-x86_64_aarch64-linux-gnu.tar.xz
+    $ pip3 install mu_environment
+    $ popd
+    ```
+
+1) Clone all repositories into the same directory as the imx-iotcore repo.
```
Comment (jordanrh1) [inline Documentation/build-arm64-firmware.md:27]: The direction I think the IMX firmware build system should go is to use submodules for deterministic versioning and a Makefile to automate the build process, like I did for LSDK (https://github.com/ms-iot/lsdk).

**Review by jordanrh1 — APPROVED** [review]

File: Documentation/build-arm64-firmware.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+       |- imx-mkimage
+       |- ms-tpm-20-ref
+       |- mu_platform_nxp
+       |- optee_os
+    ```
+
+1) Build firmware to test the setup. Adding "-j 20" to make will parallelize the build and speed it up significantly on WSL, but since the firmwares build in parallel it will be more difficult to diagnose any build failures. You can customize the number to work best with your system.
+
+    ```bash
+    # u-boot
```
Comment (hbeberman) [inline Documentation/build-arm64-firmware.md]:
```suggestion
    # U-Boot
```

File: Documentation/build-arm64-firmware.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    export CROSS_COMPILE=~/gcc-linaro-7.2.1-2017.11-x86_64_aarch64-linux-gnu/bin/aarch64-linux-gnu-
+    export ARCH=arm64
+    
+    pushd u-boot
+    make imx8mq_evk_nt_defconfig
+    make
+    popd
+    
+    
+    # Atf
```
Comment (hbeberman) [inline Documentation/build-arm64-firmware.md]:
```suggestion
    # Arm Trusted Firmware
```

File: Documentation/build-arm64-firmware.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+    # make PLATFORM=imx PLATFORM_FLAVOR=mx8mqevk \
+    #  CFG_TEE_CORE_DEBUG=y CFG_TEE_CORE_LOG_LEVEL=3  CFG_UART_BASE=0x30890000 \
+    #  CFG_RPMB_FS=y CFG_RPMB_TESTKEY=y CFG_RPMB_WRITE_KEY=y CFG_REE_FS=n \
+    #  CFG_TA_DEBUG=y CFG_TEE_CORE_TA_TRACE=1 CFG_TEE_TA_LOG_LEVEL=2 \
+    #  CFG_IMXCRYPT=y CFG_CORE_HEAP_SIZE=98304
+    
+    ${CROSS_COMPILE64}objcopy -O binary ./out/arm-plat-imx/core/tee.elf ./out/arm-plat-imx/tee.bin
+    popd
+    
+    # OPTEE- Tas
```
Comment (hbeberman) [inline Documentation/build-arm64-firmware.md]:
```suggestion
    # OP-TEE Trusted Applications
```

File: Documentation/build-arm64-firmware.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+    # Atf
+    
+    export CROSS_COMPILE=~/gcc-linaro-7.2.1-2017.11-x86_64_aarch64-linux-gnu/bin/aarch64-linux-gnu-
+    export ARCH=arm64
+    
+    pushd imx-atf
+    make PLAT=imx8mq SPD=opteed bl31
+    popd
+    
+    # Optee-OS
```
Comment (hbeberman) [inline Documentation/build-arm64-firmware.md]:
```suggestion
    # OP-TEE OS
```

**Review by hbeberman — APPROVED** [review]

**Pull request merged**